src/vs/vscode.proposed.d.ts

```diff
@@ -33,7 +33,37 @@ declare module 'vscode' {
 		extensionHostEnv?: { [key: string]: string | null };
 	}
 
-	export type ResolverResult = ResolvedAuthority & ResolvedOptions;
+	export interface Port extends Disposable {
+		remotePort: number;
```

Comment:
Not sure if this is will matter for remote or not but one thing for the VSO embedder is that the `localPort` would actually be `https` address instead of just a port number

Reply:
On the web, there is no resolver executing, so VSO does not interact / will not interact with this API because they implement port forwarding through the embedder API, not through extension API.

Reply:
+1 for calling it `Tunnel` instead.